packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1803,6 +1827,73 @@ abstract class DiagnosticsNode {
     }());
     return result;
   }
+
+  void _jsonifyNextNodesInStack(
+    _NodesToJsonifyStack toJsonify, {
+    required DiagnosticsSerializationDelegate delegate,
+  }) {
+    while (toJsonify.isNotEmpty) {
+      final (
+        DiagnosticsNode nextNode,
+        void Function(_JsonDiagnosticsNode) callback
+      ) = toJsonify.removeAt(0);
```

Comment:
if we are always removing from the front, should we use a more efficient data structure than a list like a ListQueue?

Reply:
Ooh I didn't know ListQueue exists, that's perfect